token_utils/unpack.py

```diff
@@ -0,0 +1,46 @@
+"""
+Token unpacking and verifying functions.
+"""
+
+from time import time
+
+from django.conf import settings
+from jwkest import Expired, Invalid, jwk
+from jwkest.jws import JWS
+
+
+def unpack_jwt(token, lms_user_id, now=None):
+    """
+    Unpack and verify an encoded JWT.
+
+    Validate the user, expiration, and block.
+
+    Arguments:
+        token (string): The token to be unpacked and verified.
+        lms_user_id (int): LMS user ID this token should match with.
+        now (int): Optional now value for testing.
+    """
+    now = now or int(time())
+    payload = _unpack_and_verify(token)
+
+    # todo: are these the correct exceptions to throw?
+    # todo: should all relevant exceptions be thrown instead of just 1?
+    if payload["lms_user_id"] != lms_user_id:
```

Comment:
Question: should I also include a check that these fields actually exist (lms_user_id and exp)?

Reply:
what do you get now if the payload has no "exp"? Would it be nicer to have a cleaner error about missing fields?

Reply:
Yes, that would be better. I've included error handling for that and updated tests :)